common4j/src/main/com/microsoft/identity/common/java/nativeauth/controllers/results/INativeAuthCommandResult.kt

```diff
@@ -62,7 +70,19 @@ interface INativeAuthCommandResult {
         SignUpSubmitCodeCommandResult, SignUpResendCodeCommandResult,
         SignUpSubmitPasswordCommandResult,
         ResetPasswordStartCommandResult, ResetPasswordSubmitCodeCommandResult,
-        ResetPasswordResendCodeCommandResult, ResetPasswordSubmitNewPasswordCommandResult
+        ResetPasswordResendCodeCommandResult, ResetPasswordSubmitNewPasswordCommandResult {
+        override fun toSafeString(mayContainPii: Boolean): String {
+            return if (mayContainPii) {
+                "UnknownError(correlationId=$correlationId, error=$error, errorDescription=$errorDescription), details=$details, errorCodes=$errorCodes"
+            } else {
+                "UnknownError(correlationId=$correlationId)"
+            }
+        }
+
+        override fun containsPii(): Boolean = true
+
+        override fun toString(): String = toSafeString(false)
```

Comment:
The `toString()` method should use `toSafeString(true)` as that would mean that any accidental use of this method does not contain any PII data. If anyone wants a String object with PII then they explicitly call the _dangerous_ method.

Reply:
I believe I answered this above (i.e. a different meaning of `mayContainPii`). Let me know if it doesn't.

Reply:
Please resolve if addressed @SaurabhMSFT